.pre-commit-config.yaml

```diff
@@ -0,0 +1,11 @@
+repos:
+  - repo: https://github.com/google/yapf
+    rev: v0.32.0  # Keep the same as requirements-dev.txt
+    hooks:
+    -   id: yapf
+        args: ['-i', '--recursive', '--parallel', '--exclude build/**']
+        additional_dependencies: [toml]
+  - repo: https://github.com/pylint-dev/pylint
+    rev: v2.8.2  # Keep the same as requirements-dev.txt
+    hooks:
+      - id: pylint
```

Comment:
I am afraid that we need much more tools than this, because we will need to check both Python and C++ code. In addition, vLLM has Markdown files, yaml configs, and others. Here is what I am exploring

```yaml
---
fail_fast: false
repos:
    - repo: https://github.com/pre-commit/pre-commit-hooks
      rev: v4.4.0
      hooks:
          - id: trailing-whitespace
          - id: end-of-file-fixer
    - repo: https://github.com/executablebooks/mdformat
      rev: 0.7.16
      hooks:
          - id: mdformat
            additional_dependencies:
                - mdformat-gfm
                - mdformat-black
    - repo: https://github.com/cheshirekow/cmake-format-precommit
      rev: v0.6.13
      hooks:
          - id: cmake-format
    - repo: https://github.com/shellcheck-py/shellcheck-py
      rev: v0.9.0.5
      hooks:
          - id: shellcheck
            args: [--severity=error]
    - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
      rev: 0.2.2
      hooks:
          - id: yamlfmt
    - repo: https://github.com/pre-commit/mirrors-mypy
      rev: v1.3.0
      hooks:
          - id: mypy
    - repo: https://github.com/psf/black
      rev: 23.3.0
      hooks:
          - id: black
            args: [--line-length=100]
    - repo: https://github.com/pycqa/isort
      rev: 5.12.0
      hooks:
          - id: isort
            name: isort (python)
    - repo: https://github.com/PyCQA/flake8
      rev: 6.0.0
      hooks:
          - id: flake8
    - repo: https://github.com/pocc/pre-commit-hooks
      rev: v1.3.5
      hooks:
          - id: clang-format
            args: [--style=Google]
          # - id: clang-tidy # No easy way to install clang-tidy on macOS
          # - id: oclint # oclint uses dylds that were not verified by macOS
          # - id: uncrustify # seems conflicts with clang-format
          - id: cppcheck
          - id: cpplint
          # - id: include-what-you-use # Cannot handle including files from external frameworks
```

Reply:
I think for markdown there is no need to.

For python iirc there is also a pr moving this to ruff. lets just do that. 

Lets not use black here, instead yapf should be used since the style is currently conform to yapf.